examples/web/index.js

```diff
@@ -0,0 +1,87 @@
+/* eslint gigwalk/require-valid-flow-file-annotation: 0 */
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import injectTapEventPlugin from 'react-tap-event-plugin';
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import TextField from 'material-ui/TextField';
+import AppBar from 'material-ui/AppBar';
+import RaisedButton from 'material-ui/RaisedButton';
+import Gigwalk from '../../lib/Client';
+
+injectTapEventPlugin();
+
+const client = new Gigwalk();
+
+const formStyle = {
+    margin: '0 auto',
+    maxWidth: 1000,
+    paddingLeft: 40,
+    paddingRight: 40
+};
+
+class App extends Component {
+    static propTypes = {
+        client: React.PropTypes.shape({
+            authorize: React.PropTypes.func.isRequired
+        }).isRequired
+    };
+    constructor(...args) {
+        super(...args);
+        this.state = {
+            username: '',
+            password: '',
+            res: ''
+        };
+        this.onChangeEmail = this.onChangeEmail.bind(this);
+        this.onChangePassword = this.onChangePassword.bind(this);
+        this.onSubmit = this.onSubmit.bind(this);
+    }
+
+    onChangeEmail(e) {
+        this.setState({
+            username: e.target.value
+        });
+    }
+
+    onChangePassword(e) {
+        this.setState({
+            password: e.target.value
+        });
+    }
+    onSubmit() {
+        this.props.client.authorize(this.state.username, this.state.password);
```

Comment:
This is an example of how the queue will prevent duplicate requests and will just return the same promise.